### ISSUE TITLE
Out-of-range indication don't work with #showtooltip (without a specific spell name after it)

### ISSUE THREAD

**Issue by Rollo-Tomasi-git** [bhhandley/CleveRoidMacros#5]
This doesn't give me a "red overlay" on the button when I'm out of range:
#showtooltip
/retarget
/stopcasting
/startattack
/cast Throw

This one does:
#showtooltip Throw
/retarget
/stopcasting
/startattack
/cast Throw

(The reason this is a problem is I want to make a general "throw / shoot xxx" macro that detects the kind of ranged weapon I'm using.)

Note: I'm using Bongos, maybe that matters?

PS: I love your work on this addon, thank you! :-D

**Comment by bhhandley** [comment]
Pushed a fix for this, please test again and let me know.

**Comment by Rollo-Tomasi-git** [comment]
Nope, didn't fix it. It actually got worse, I can't use #showtooltip Throw either now. But if I skip #showtooltip alltogether, it works:
/retarget
/stopcasting
/startattack
/cast Throw

However, what I'm actually trying to do is this: (which doesn't show range-indication atm)
#showtooltip
/retarget
/stopcasting
/startattack
/cast [equipped:Thrown] Throw; [equipped:Bows] Shoot Bow; [equipped:Guns] Shoot Gun; Shoot Crossbow

**Comment by bhhandley** [comment]
Apologize for the delay.  I have a new version coming out hopefully in the next couple days, just waiting for an update to nampower.  This is fixed -- I hadn't realized that superwow was also attempting to update various things which had thrown off what I was doing.  It will still always be better with superwow and nampower (sometimes required).  With the update, if you're not using those mods, as long as you have at least one ability anywhere on your bars with the same min/max range as what you're trying to use, it should give a proper range indication.  For highlighting (again without SW/NP) a reactive spell would always need to be on your bars somewhere and the same for auto-repeat or on-swing spells.  There's just no other way to get that information without the mods other than to check the ability from the actionbars.

If you have superwow and no conditionals (like `/cast Throw`) or use the older trick of putting `/script -- CastSpellByName("Throw")` at the top, it will try to update the icon.  So, is/was a little conflict going on that I hadn't realized.

**Comment by Kissabella** [comment]
Hello,
I have the same problem!
I have it with Judgment, I use pfUI (absolutely perfect addon), I use #showtooltip in my macro

#showtooltip
/startattack
/cast [@mouseover,harm,nodead] Judgment; Judgment

The macro shows the normal tooltip and CD, but not the range indicator and the “Ready” indicator. For example, "Judgment" is displayed as normal, even if no "Seal" is activated, it is not grayed out, even if the target is out of range, then the icon is not grayed out.

And it's also when other attack skills don't have a target that they won't be grayed out.

I tested it with with/without pfUI, with/without mouserovercast, with/without /startattack, tooltip is not displayed correctly

**Comment by bhhandley** [comment]
> Hello, I have the same problem! I have it with Judgment, I use pfUI (absolutely perfect addon), I use #showtooltip in my macro
> 
> #showtooltip /startattack /cast [@mouseOver,harm,nodead] Judgment; Judgment
> 
> The macro shows the normal tooltip and CD, but not the range indicator and the “Ready” indicator. For example, "Judgment" is displayed as normal, even if no "Seal" is activated, it is not grayed out, even if the target is out of range, then the icon is not grayed out.
> 
> And it's also when other attack skills don't have a target that they won't be grayed out.
> 
> I tested it with with/without pfUI, with/without mouserovercast, with/without /startattack, tooltip is not displayed correctly

Do you have nampower installed?

**Comment by Kissabella** [comment]
no

**Comment by bhhandley** [comment]
Then you have two options that I mentioned in my previous post.  

1. Install nampower.  There will also be an update to nampower coming soon which is what my next release is waiting on.
2. If you are using superwow then remove `#showtooltip` and add `/run -- CastSpellByName("Judgement")` this should allow superwow to handle it instead.

If you don't have nampower or superwow then there's currently no way for the button to react properly until the next release.  As I mentioned in my previous post.

**Comment by Thaodan** [comment]
I noticed similarly the  indicator of specific spell to be active before the icon can  be not greyed out doesn't work.
I.e. if I have a judgement macro.
